cli/command/service/update.go

```diff
@@ -364,6 +374,27 @@ func updateEnvironment(flags *pflag.FlagSet, field *[]string) {
 	*field = removeItems(*field, toRemove, envKey)
 }
 
+func getUpdatedSecrets(apiClient client.APIClient, flags *pflag.FlagSet, secrets []*swarm.SecretReference) ([]*swarm.SecretReference, error) {
```

Comment:
are we not doing a `target` conflict check here? The API will error out, but seems like we should error early?